src/Components/ArticlePage.jsx

```diff
@@ -1,37 +1,47 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+import { getArticleById } from "../api/api";
+import Error from "./Error";
 
 const ArticlePage = () => {
   const { id } = useParams();
-  console.log(useParams());
-  const [article, setArticle] = useState(null);
+  const [article, setArticle] = useState({});
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [isError, setIsError] = useState(false);
```

Comment:
I think the state for `isError` isn't necessary, you have a `null` value for `error` when there isn't an error and you can use the falsey nature of the `null` value in any conditional logic